### ISSUE TITLE
Swapped order for if/else with early returns

### ISSUE THREAD

**Issue by simonlindholm** [matt-kempster/mips_to_c#88]
mips_to_c can give output like
```c
        temp_v0 = D_800D6FE8.buttonHeld;
        if ((temp_v0 & 0x400) == 0) {
            if ((D_800D6FE8.buttonPressed & 0x8000) != 0) {
                if ((((D_8004A7C4->objId * 4) + 0x800F0000)->unk-7520 & 6) != 6) {
block_6:
                    set_action_1(3, 5);
                    return 2;
                }
                if ((temp_v0 & 0xB00) == 0) {
                    goto block_6;
                }
                gState.unk44 = 1;
                set_action_1(0x17, 0x1B);
                return 2;
            }
        }
```
which looks like it fails to identify a ||, but in fact it's an && and the if/else blocks are reversed. We probably detect early returns too eagerly.

**Comment by zbanks** [comment]
It looks like this was fixed by #114.
Before:
```c
void func_80C06B70(void *arg0, ?32 arg1) {
    s16 sp26;
    s16 temp_v1;
    s32 phi_v0;

    // Node 0
    temp_v1 = arg0->unk92 - arg0->unk32;
    sp26 = temp_v1;
    if (func_800B84D0(arg0, arg1) != 0) {
        // Node 1
        if ((s32) arg0->unk38 >= 0) {
            // Node 3
            if (ActorCutscene_GetCurrentIndex(arg0) == 0x7C) {
                // Node 4
                ActorCutscene_Stop(0x7C);
            }
            // Node 5
            ActorCutscene_SetIntentToPlay(arg0->unk38);
            func_80C06DC8(arg0, arg1);
            func_80C06C54(arg0);
            return;
        }
        // Node 2
        func_80C06DC8(arg0, arg1);
        func_80C06CC4(arg0);
        return;
    }
    // Node 6
    phi_v0 = (s32) temp_v1;
    if ((s32) temp_v1 < 0) {
        // Node 7
        phi_v0 = -(s32) temp_v1;
    }
    // Node 8
    if (phi_v0 < 0x2AAA) {
        // Node 9
        func_800B8614(arg0, arg1, 0x42A00000);
    }
    // Node 10
}
```
After:
```c
void func_80C06B70(void *arg0, ?32 arg1) {
    s16 sp26;
    s16 temp_v1;
    s32 phi_v0;

    // Node 0
    temp_v1 = arg0->unk92 - arg0->unk32;
    sp26 = temp_v1;
    if (func_800B84D0(arg0, arg1) != 0) {
        // Node 1
        if ((s32) arg0->unk38 < 0) {
            // Node 2
            func_80C06DC8(arg0, arg1);
            func_80C06CC4(arg0);
            // Node 10.1
            return;
        }
        // Node 3
        if (ActorCutscene_GetCurrentIndex(arg0) == 0x7C) {
            // Node 4
            ActorCutscene_Stop(0x7C);
        }
        // Node 5
        ActorCutscene_SetIntentToPlay(arg0->unk38);
        func_80C06DC8(arg0, arg1);
        func_80C06C54(arg0);
        // Node 10.2
        return;
    }
    // Node 6
    phi_v0 = (s32) temp_v1;
    if ((s32) temp_v1 < 0) {
        // Node 7
        phi_v0 = -(s32) temp_v1;
    }
    // Node 8
    if (phi_v0 < 0x2AAA) {
        // Node 9
        func_800B8614(arg0, arg1, 0x42A00000);
    }
    // Node 10
}
```

**Comment by simonlindholm** [comment]
I wouldn't be surprised if we see more cases of this, but let's close it until that point.
